src/github.com/stellar/gateway/compliance/handlers/main.go

```diff
@@ -18,4 +21,21 @@ type RequestHandler struct {
 	SignatureSignerVerifier crypto.SignerVerifierInterface `inject:""`
 	StellarTomlResolver     stellartoml.ResolverInterface  `inject:""`
 	FederationResolver      federation.ResolverInterface   `inject:""`
+	NonceGenerator          NonceGeneratorInterface        `inject:""`
+}
+
+type NonceGeneratorInterface interface {
+	Generate() string
+}
+
+type NonceGenerator struct{}
+
+func (n *NonceGenerator) Generate() string {
+	return strconv.FormatInt(time.Now().UnixNano(), 10)
```

Comment:
I don't think we need secure random here (but sure @stanford-scs should check it out). We just need two identical attachments (minus `nonce`) to have different hashes.